### PR TITLE
Namespace API over REST v2 - fetch HEAD if hashOnRef is not specified

### PR DESCRIPTION
**Pull request by snazy**

### CODE REVIEW

**Review by dimas-b — APPROVED** [review]
Did you hit a test failure for this somewhere?

File: clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideCreateNamespace.java [hunk 1]
```diff
@@ -74,11 +74,16 @@ public Namespace create()
     }
 
     try {
+      String expectedHash = hashOnRef;
+      if (expectedHash == null) {
+        expectedHash = api.getReference().refName(refName).get().getHash();
```
Comment (dimas-b) [inline clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideCreateNamespace.java:79]: Could you add a test case for this?

Comment (snazy) [reply clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideCreateNamespace.java:79]: THat's coming w/ #5924

**Review by dimas-b — CHANGES_REQUESTED** [review]
Just realized that we should probably have a test case for this because I updated old tests to provide expected hashes in #5828

File: clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideCreateNamespace.java (hunk 1 above)
Comment (dimas-b) [reply clients/client/src/main/java/org/projectnessie/client/util/v2api/ClientSideCreateNamespace.java:79]: cool :+1:

**Pull request merged**